common/cache/lru.go

```
@@ -181,26 +184,24 @@ func New(opts *Options) Cache {
 		cache.metricsScope = metrics.NoopScope(1)
 	}
 
-	if cache.isSizeBased {
-		cache.sizeFunc = opts.GetCacheItemSizeFunc
-		cache.maxSize = opts.MaxSize
-		if cache.maxSize == nil {
-			// If maxSize is not defined for size-based cache, set default to cacheCountLimit
-			cache.maxSize = dynamicproperties.GetIntPropertyFn(cacheDefaultSizeLimit)
-		}
-		cache.sizeByKey = make(map[interface{}]uint64, opts.InitialCapacity)
+	if opts.IsSizeBased == nil {
+		cache.isSizeBased = dynamicproperties.GetBoolPropertyFn(false)
```

I prefer to panic in this case. I'd like to have the behavior of cache to be completely managed by dynamic config. These implicit logic can make it hard to debug

---

I want to add the panic logic later once we add feature flags for all the cache usages. For now, existing usage will be default to count based.

---

I'd avoid panic'ing in library level and instead return an error for missing mandatory options

---

Thanks, I will add the error returning and make this into a mandatory flag once we add the feature flags for all other cache usage.